src/classes/PermanentFileSystem.ts

```diff
@@ -54,6 +54,20 @@ const isItemPath = (requestedPath: string): boolean => (
   && requestedPath.split('/').length > 3
 );
 
+// TODO: this exists defensively, and maybe should continue to exist defensively,
+// but in theory the backend will stop returning duplicates at some point.
+// See https://github.com/PermanentOrg/back-end/issues/396
+const deduplicateFileEntries = (fileEntries: FileEntry[]): FileEntry[] => (
+  fileEntries.filter(
+    (filterFileEntry, index, self) => {
```

Comment:
This function is more complex than I would want it to be.

Given more time I would prefer it:

1. Be in a utility file so it could be unit tested (eventually)
2. Have these sub-functions as non-exported named helper functions (e.g. `fileNameFilter` and `getFirstIndexByFilename`) to break up the nested-and-kinda-complicated array operations.

Maybe I'll do that despite the time ;)

Reply:
Since you are here already....

`duplicateFileEntries` seem to suggest that the opposite of what that function is doing. Something like `removeDuplicateCopyFileEntries` or similar?

Reply:
It's `deduplicate`, all good on that, sorry for the dust!